### PR TITLE
feat(ui): add FirebaseTable & FirebaseListView

### PR DESCRIPTION
**Pull request by rrousselGit**
## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/flutterfire_ui/example/web/index.html [hunk 1]
```diff
@@ -25,9 +25,11 @@
 
     <title>flutterfire_ui_example</title>
     <link rel="manifest" href="manifest.json" />
-    <script src="https://www.gstatic.com/firebasejs/8.6.1/firebase-app.js"></script>
-    <script src="https://www.gstatic.com/firebasejs/8.6.1/firebase-auth.js"></script>
-    <script src="https://www.gstatic.com/firebasejs/8.6.1/firebase-firestore.js"></script>
+    <script src="https://www.gstatic.com/firebasejs/8.10.0/firebase-app.js"></script>
+    <script src="https://www.gstatic.com/firebasejs/8.10.0/firebase-auth.js"></script>
+    <script src="https://www.gstatic.com/firebasejs/8.10.0/firebase-firestore.js"></script>
+    <script src="https://www.gstatic.com/firebasejs/8.10.0/firebase-firestore.js"></script>
```
Comment (lesnitsky) [inline packages/flutterfire_ui/example/web/index.html:31]: duplicate entry

File: packages/flutterfire_ui/lib/src/firebase_query_builder.dart [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+    Key? key,
+    required Query query,
+    required Widget Function(
+      BuildContext context,
+      DataSnapshot doc,
+    )
+        itemBuilder,
+    int pageSize = 10,
+    Widget Function(BuildContext context)? loadingBuilder,
+    Widget Function(
```
Comment (lesnitsky) [inline packages/flutterfire_ui/lib/src/firebase_query_builder.dart:404]: does it make sense to extract these to typedefs?

File: packages/flutterfire_ui/example/web/index.html [hunk 3]
```diff
@@ -25,9 +25,11 @@
 
     <title>flutterfire_ui_example</title>
     <link rel="manifest" href="manifest.json" />
-    <script src="https://www.gstatic.com/firebasejs/8.6.1/firebase-app.js"></script>
-    <script src="https://www.gstatic.com/firebasejs/8.6.1/firebase-auth.js"></script>
-    <script src="https://www.gstatic.com/firebasejs/8.6.1/firebase-firestore.js"></script>
+    <script src="https://www.gstatic.com/firebasejs/8.10.0/firebase-app.js"></script>
+    <script src="https://www.gstatic.com/firebasejs/8.10.0/firebase-auth.js"></script>
+    <script src="https://www.gstatic.com/firebasejs/8.10.0/firebase-firestore.js"></script>
```
Comment (rrousselGit) [inline packages/flutterfire_ui/example/web/index.html:30]:
```suggestion
```

File: packages/flutterfire_ui/example/web/index.html (hunk 1 above)
Comment (rrousselGit) [reply packages/flutterfire_ui/example/web/index.html]: Ah, merge issue. Fix

**Review by lesnitsky — APPROVED** [review]

**Pull request merged**